src/classes/CW_AudienceCSVProcessHelper.cls

```diff
@@ -0,0 +1,406 @@
+	   		//rewRec.Reward_Catalogue__c=rec.Reward_Library__r.Reward_Catalogue__c;
+	   		//rewRec.Reward_Curency__c=rec.Reward_Library__r.Currency__c;
+	   		CampRewList.Add(rewRec);
+	   		campFreeBetdMap.put(FreeBet,rewRec);
+   		}
+   	}
+   	insert CampRewList;
+   	for(CW_Campaign_Reward__c rec : CampRewList){
+   		campRewardMap.put(rec.Reward_Library__c,rec);
+   	}
+   }
+   
+   
+   private void  createLoadAudienceGroupSchdule(){
+   	
+   	grupSheduleMap = new Map<ID,CW_Audience_Group_Schedule__c>();//Reference by group ID.
+   	Map<Id,CW_Audience_Group__c> grupMap= new Map<Id,CW_Audience_Group__c>();
+   	CW_Audience_Group_Schedule__c c = new  CW_Audience_Group_Schedule__c();
+   	for(CW_CSV_Stage_AudienceMembers__c rec:CsvStageList ){
+   		if(!grupSheduleMap.containskey(rec.Audience_Group_ref__c)){
+   			grupSheduleMap.Put(rec.Audience_Group_ref__c,null);
+   			grupMap.put(rec.Audience_Group_ref__c, new CW_Audience_Group__c(id=rec.Audience_Group_ref__c,
+   											is_control_group__c=rec.Audience_Group_ref__r.Is_control_group__c,
+   											control_group__c=rec.Audience_Group_ref__r.control_group__c));
+   		}
+   	}
+   	for (CW_Audience_Group_Schedule__c rec: [SELECT ID,Audience_Group__c FROM CW_Audience_Group_Schedule__c
+   													WHERE Schedule__c=:CsvStatus.Schedule__c ]){ //Audience_Group__c=:grupSheduleMap.KeySet()  AND  load  all group schedule to set ref to control grouo sched.
+   		grupSheduleMap.put(rec.Audience_Group__c,rec);
+   														
+   	}
+   	list<CW_Audience_Group_Schedule__c> ListToIns=new list<CW_Audience_Group_Schedule__c>();
+   	for(Id grpId:grupMap.KeySet()){
+   		CW_Audience_Group_Schedule__c rec=grupSheduleMap.get(grpId);
+   		if(rec==null){
+   			rec=new CW_Audience_Group_Schedule__c();
+   			CW_Audience_Group__c grpRec=grupMap.get(grpId);
+   	 		rec.Schedule__c=CsvStatus.Schedule__c;
+   	 		rec.Is_Control_Group__c=grpRec.Is_Control_Group__c;
+   	 		rec.Audience_Group__c=grpId;
+   	 		rec.Exclude_From_RollUp__c=false;
+   	 		rec.Control_Group_Schedule__c=grprec.Control_Group__c;
+   	 		if(rec.Control_Group_Schedule__c!=null){
+   	 			CW_Audience_Group_Schedule__c CtrSched=grupSheduleMap.get(grprec.Control_Group__c);
+   	 			if(CtrSched!=null){
+   	 				rec.Control_Group_Schedule__c=CtrSched.Id;
+   	 			}
+   	 		}
+   	 		grupSheduleMap.put(grpId,rec);
+   	 		ListToIns.Add(rec);
+   		}
+   	}
+   	insert ListToIns;
+   }
+  
+   //This is complicated so we must process both to not make to much of shcedureRewards
+   //We have one shedule so we must find grup and reward(freebet)
+   private void createAudinceGroupScheduleReward(){
+ 
+   	//There a many so we must have coposit key grupId _FreeBetId (Campaign Reward)
+   	audSchedRewMap= new map<string,id>();
+   	for(CW_Audience_Group_Schedule_Reward__c rec :[SELECT Id,Audience_Group_Schedule__r.Audience_Group__c,Campaign_Reward__c FROM CW_Audience_Group_Schedule_Reward__c 
```

Comment:
Problem: Unbound SOQL query. Reason: a SOQL query can return up to 50,000 records. If this number is exceeded a runtime exception is thrown. Suggestion: cap the max number of records returned adding a LIMIT clause to the query.